synapseclient/models/mixins/table_operator.py

```diff
@@ -0,0 +1,2836 @@
+            ```
+
+        Example: Deleting a column (async)
+            This example shows how you may delete a column from a table and then store
+            the change back in Synapse.
+
+            ```python
+            import asyncio
+            from synapseclient import Synapse
+            from synapseclient.models import Table
+
+            syn = Synapse()
+            syn.login()
+
+            async def main():
+                table = await Table(
+                    id="syn1234"
+                ).get_async(include_columns=True)
+
+                table.delete_column(name="my_column")
+                table.store_async()
+
+            asyncio.run(main())
+            ```
+        """
+        if not self._last_persistent_instance:
+            raise ValueError(
+                "This method is only supported after interacting with Synapse via a `.get()` or `.store()` operation"
+            )
+        if not self.columns:
+            raise ValueError(
+                "There are no columns. Make sure you use the `include_columns` parameter in the `.get()` method."
+            )
+
+        column_to_delete = self.columns.get(name, None)
+        if not column_to_delete:
+            raise ValueError(f"Column with name {name} does not exist in the table.")
+
+        self._columns_to_delete[column_to_delete.id] = column_to_delete
+        self.columns.pop(column_to_delete.name, None)
+
+    def add_column(
+        self, column: Union["Column", List["Column"]], index: int = None
+    ) -> None:
+        """Add column(s) to the table. Note that this does not store the column(s) in
+        Synapse. You must call the `.store()` function on this table class instance to
+        store the column(s) in Synapse. This is a convenience function to eliminate
+        the need to manually add the column(s) to the dictionary.
+
+
+        This function will add an item to the `.columns` attribute of this class
+        instance. `.columns` is a dictionary where the key is the name of the column
+        and the value is the Column object.
+
+        Arguments:
+            column: The column(s) to add, may be a single Column object or a list of
+                Column objects.
+            index: The index to insert the column at. If not passed in the column will
+                be added to the end of the list.
+
+        Returns:
+            None
+
+        Example: Adding a single column
+            This example shows how you may add a single column to a table and then store
+            the change back in Synapse.
+
+            ```python
+            from synapseclient import Synapse
+            from synapseclient.models import Column, ColumnType, Table
+
+            syn = Synapse()
+            syn.login()
+
+            table = Table(
+                id="syn1234"
+            ).get(include_columns=True)
+
+            table.add_column(
+                Column(name="my_column", column_type=ColumnType.STRING)
```

Comment:
It looks like the `name` parameter might have gotten changed to `column` at some point?